dsgrid/query/models.py

```diff
@@ -200,10 +205,56 @@ class ProjectQueryParamsModel(DSGBaseModel):
         description="Controls table format",
         default=TableFormatType.PIVOTED,
     )
+
+    @validator("dimension_filters")
+    def handle_dimension_filters(cls, dimension_filters):
+        for i, dimension_filter in enumerate(dimension_filters):
+            if not isinstance(dimension_filter, DimensionFilterBaseModel):
+                dimension_filters[i] = make_dimension_filter(dimension_filter)
+        return dimension_filters
+
+    @validator("per_dataset_aggregations")
+    def check_aggregations(cls, aggregations):
+        for agg in aggregations:
+            for dim_type in DimensionType:
+                field = dim_type.value
+                for item in getattr(agg.dimensions, field):
+                    # TODO: we could add support for this, but it does add some ambiguity.
+                    if item.function is not None:
+                        raise ValueError(
+                            f"function={item.function} cannot be set in ProjectQueryParamsModel"
+                        )
+                    if item.alias is not None:
+                        # TODO: need to support aliases here. The code that handles columns during
+                        # dataset concatenation doesn't support it.
+                        raise ValueError(
+                            f"alias={item.alias} cannot be set in ProjectQueryParamsModel"
+                        )
+        return aggregations
+
+
+class ProjectQueryParamsModel(CacheableQueryBaseModel):
+    """Defines how to transform a project into a CompositeDataset"""
+
+    project_id: str = Field(description="Project ID for query")
+    dataset_ids: List[str] = Field(description="Dataset IDs from which to read")
+    excluded_dataset_ids: List[str] = Field(
+        description="Datasets to exclude from query", default=[]
+    )
+    # TODO: default needs to change
+    include_dsgrid_dataset_components: bool = Field(description="", default=False)
+    dataset_params: ProjectQueryDatasetParamsModel = Field(
+        description="Parameters affecting datasets. Will be used for caching intermediate tables.",
+        default=ProjectQueryDatasetParamsModel(),
+    )
```

Comment:
Is this additional layer of nesting necessary/helpful? It seems like a lot of what was in ProjectQueryParamsModel is now in ProjectQueryDatasetParamsModel, but you haven't added a whole lot to ProjectQueryParamsModel, and ProjectQueryDatasetParamsModel doesn't appear to be per-dataset information (whereas the spark_conf_per_dataset is, but that's achieved by providing a list of objects that identify the dataset_id).

Reply:
I added ProjectQueryDatasetParamsModel to enable caching. The three parameters in that class uniquely identify a project-mapped dataset. I agree that it doesn't help the organizational structure.